addons/ttf/ttf.c

```diff
@@ -1005,10 +1005,10 @@ static bool ttf_get_glyph_dimensions(ALLEGRO_FONT const *f,
       }
    }
    cache_glyph(data, face, ft_index, glyph, false);
-   *bbx = glyph->offset_x;
+   *bbx = glyph->offset_x + 1;
    *bbw = glyph->region.w - 2;
```

Comment:
I have tested this extensively and I am sure I did make a mistake by returning glyph->offset_x+1. That's not right. The value "glyph->offset_x" is exactly the position where the glyph begins. You can see in [these lines](https://github.com/liballeg/allegro5/blob/master/addons/ttf/ttf.c#L521-L528) of ttf.c that the glyph is drawn with offsets exactly (offset_x, offset_y) (they grab the glyph image without including the 1 pixel margin). 

For example, this code:
```c
    char t = 'a';
    int x = 1;
    al_get_glyph_dimensions(default_font, t, &bbx, &bby, &bbw, &bbh);
    al_draw_glyph(default_font, WHITE_COLOR, x, x, t);
    x+=0.5; // to get pixel-aligned rectangle
    al_draw_rectangle(x+bbx, x+bby, x+bbx+bbw-1, x+bby+bbh-1, al_map_rgba(255,0,0,140), 1);
```
Produces this with the current version (after this patch):
![ex_current](https://cloud.githubusercontent.com/assets/460962/15994143/6dc8e8f0-30d2-11e6-9439-dd6ad8a4f6dd.png)

And this is what it looks like if one fixes the return values of bbx and bby (as glyph->offset_x without adding anything):
![ex_shouldbe](https://cloud.githubusercontent.com/assets/460962/15994156/085ac10e-30d3-11e6-8b7d-bf485884eb1f.png)